### ISSUE TITLE
 Is this project to be compiled on WSL and then run on winxp? Have you ever thought of developing on VS2019 directly

### ISSUE THREAD

**Issue by shadowbanewuha** [etorth/mir2x#36]

**Comment by shadowbanewuha** [comment]
`
[ 20%] Built target sol2
[ 22%] Built target SQLiteCpp
[ 24%] Built target tinyxml2
[ 26%] Built target utf8cpp
[ 28%] Built target libzstd
[ 28%] Built target mir2x_3rds
[ 33%] Built target common
Consolidate compiler generated dependencies of target monoserver

[ 33%] Linking CXX executable monoserver
ld: library not found for -lXrender
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [server/src/monoserver] Error 1
make[1]: *** [server/src/CMakeFiles/monoserver.dir/all] Error 2
make: *** [all] Error 2

AltaideMacBook-Pro:b altai$ brew install libxrender
Error: 
  homebrew-core is a shallow clone.
To `brew update`, first run:
  git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow
This command may take a few minutes to run due to the large size of the repository.
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
automatically to avoid repeatedly performing an expensive unshallow operation in
CI systems (which should instead be fixed to not use shallow clones). Sorry for
the inconvenience!
Warning: libxrender 0.9.10 is already installed and up-to-date.
To reinstall 0.9.10, run:
  brew reinstall libxrender

AltaideMacBook-Pro:b altai$ 
`

link error....

**Comment by etorth** [comment]
was able to build with vs but later the coroutine support breaks it.

**Comment by etorth** [comment]
I may consider rollback fltk to 1.3.x, I previously tried 1.4.0.

**Comment by shadowbanewuha** [comment]
Where is. wil. wix files

**Comment by etorth** [comment]
if just to run game, you don't need it.

If you are gonna run tools, you can find it here: https://github.com/etorth/CBWCQ3.git

**Comment by shadowbanewuha** [comment]
The execution time of these sh files, such as monsterwil2png.sh, monstergen.sh......
at compile time or install time or run time ?

**Comment by etorth** [comment]
You don't need to run it, it's used to generate gfx files for runtime.
All the generated file are already included when you clone the repo.

**Comment by shadowbanewuha** [comment]
{
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_progUseDB,    "root/texture/progUseDB"   ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_itemDB,       "root/texture/itemDB"      ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_mapDB,        "root/texture/mapDB"       ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_fontexDB,     "root/font/fontexDB"       ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_heroDB,       "root/texture/heroDB"      ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_hairDB,       "root/texture/hairDB"      ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_monsterDB,    "root/texture/monsterDB"   ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_weaponDB,     "root/texture/weaponDB"    ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_helmetDB,     "root/texture/helmetDB"    ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_magicDB,      "root/texture/magicDB"     ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_equipDB,      "root/texture/equipDB"     ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_standNPCDB,   "root/texture/standNPCDB"  ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_selectCharDB, "root/texture/selectCharDB"); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_mapBinDB,     "root/map/mapBinDB"        ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_emojiDB,      "root/emoji/emojiDB"       ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_bgmDB,        "root/sound/bgmDB"         ); }},
          {1, [this](size_t weight){ loadDB(weight, g_xmlConf, g_seffDB,       "root/sound/seffDB"        ); }},
      }

---------------------------------------------------------
root/texture/***DB， root/map/mapBinDB， root/emoji/emojiDB，root/sound/bgmDB，root/sound/seffDB
Where are these documents

**Comment by shadowbanewuha** [comment]
Yes. Can I generate these files myself，use these monsterwil2png.sh、magicwil2png.sh .......

**Comment by etorth** [comment]
You don't need to generate them by yourself, if you see issue loading it, means you didn't install git-lfs.
git-lfs should pull all these files automatically.

**Comment by shadowbanewuha** [comment]
I installed git lfs, but these pulled files are texture\monster. zsdb, texture\magic. zsdb....

**Comment by shadowbanewuha** [comment]
I found that zsdbmaker can decode these files and get png pictures

**Comment by shadowbanewuha** [comment]
Oh, I see.  The configuration.xml records the relationship between them

**Comment by shadowbanewuha** [comment]
I have compiled and run the moniserver now, but how can the client establish a link with it?

**Comment by etorth** [comment]
> I have compiled and run the monoserver now, but how can the client establish a link with it?

You should be able to run with default setup to connect to the server.
If not, what kind of error you have seen?

**Comment by shadowbanewuha** [comment]
load map failed; ID = 24, Name = 道馆_1

**Comment by shadowbanewuha** [comment]
MapBinDB::loadResource(24); --> Mir2xMapData .w=600, .h=0, .blockList.size=600 ;  Why is this value 0

**Comment by etorth** [comment]
I feel something is wrong for your mapdb.zsdb

**Comment by shadowbanewuha** [comment]
Seems to be the problem here
cerealf.hpp
`        T t;
        {
            cereal::PortableBinaryInputArchive ar(ss);
            ar(t);
        }`

**Comment by shadowbanewuha** [comment]
.w and .h use int64_t instead of size_t , now it can work normally.  
But there was a runtime error :  
ServerMap::ServerMapLuaModule::resumeLoop() ----- throw fflerror("ServerMap coroutine is not callable");

Is this because I use VS2019 to build？

**Comment by etorth** [comment]
Likely something wrong when serverMap executes the lua scripts.
Lua scripts are used to generate monsters, the lua coroutine get polled to advance.

Please check your log file see what's the error.

**Comment by shadowbanewuha** [comment]
Is this because VS2019 does not support coroutine？

**Comment by etorth** [comment]
> Is this because VS2019 does not support coroutine？

This is not C++ coroutine, it's lua coroutine.
If you can pass the compilation, likely you already have C++ coroutine support.

But lua coroutine looks ran into issue.
As said, did you check the log?

**Comment by shadowbanewuha** [comment]
The log does have a problem. It indicates that the lua file cannot be found, although the lua file path is correct. 
I think it may be the problem of utf8 coding or gbk coding.

Since VS supports coroutine, I will continue to built it on VS
